pkg/agent/plugin/workloadattestor/windows/windows_windows.go

```diff
@@ -0,0 +1,200 @@
+//go:build windows
+// +build windows
+
+package windows
+
+import (
+	"context"
+	"fmt"
+	"syscall"
+
+	"github.com/hashicorp/go-hclog"
+	workloadattestorv1 "github.com/spiffe/spire-plugin-sdk/proto/spire/plugin/agent/workloadattestor/v1"
+	"github.com/spiffe/spire/pkg/common/catalog"
+	"github.com/spiffe/spire/pkg/common/telemetry"
+	"golang.org/x/sys/windows"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func builtin(p *Plugin) catalog.BuiltIn {
+	return catalog.MakeBuiltIn(pluginName, workloadattestorv1.WorkloadAttestorPluginServer(p))
+}
+
+func New() *Plugin {
+	p := &Plugin{q: &processQuery{}}
+	return p
+}
+
+type Plugin struct {
+	workloadattestorv1.UnsafeWorkloadAttestorServer
+
+	log hclog.Logger
+	q   processQueryer
+}
+
+type processInfo struct {
+	pid        int32
+	user       string
+	userSID    string
+	groups     []string
+	groupsSIDs []string
+}
+
+func (p *Plugin) Attest(ctx context.Context, req *workloadattestorv1.AttestRequest) (*workloadattestorv1.AttestResponse, error) {
+	process, err := p.newProcessInfo(req.Pid, p.q)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get process information: %v", err)
+	}
+	var selectorValues []string
+	selectorValues = addSelectorValueIfNotEmpty(selectorValues, "user_name", process.user)
+	selectorValues = addSelectorValueIfNotEmpty(selectorValues, "user_sid", process.userSID)
+	for _, groupSID := range process.groupsSIDs {
+		selectorValues = addSelectorValueIfNotEmpty(selectorValues, "group_sid", groupSID)
+	}
+	for _, group := range process.groups {
+		selectorValues = addSelectorValueIfNotEmpty(selectorValues, "group_name", group)
+	}
+
+	return &workloadattestorv1.AttestResponse{
+		SelectorValues: selectorValues,
+	}, nil
+}
+
+func (p *Plugin) SetLogger(log hclog.Logger) {
+	p.log = log
+}
+
+func (p *Plugin) newProcessInfo(pid int32, q processQueryer) (*processInfo, error) {
+	h, err := q.OpenProcess(pid)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open process: %w", err)
+	}
+	defer func() {
+		if err := windows.CloseHandle(h); err != nil {
+			p.log.Warn("Could not close process handle", telemetry.Error, err)
+		}
+	}()
+
+	// Retrieve an access token to describe the security context of
+	// the process from which we obtained the handle.
+	var token windows.Token
+	err = q.OpenProcessToken(h, &token)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open the access token associated with the process: %w", err)
+	}
+	defer token.Close()
```

Comment:
This can fail too right? Should be capture the error and log a warning like the OpenProcess call above?

Reply:
Good catch!